### DISCUSSION TITLE
Small QoL Patches I've made to Poke-Env for VGC

### DISCUSSION THREAD

**Discussion by caymansimpson** [hsahovic/poke-env#269] [Show and tell]
In case any of you want to take them (from 6 months ago)
1. [Converting indexes from our internal `active_pokemon` to showdown target indexes](https://github.com/caymansimpson/poke-env/blob/master/src/poke_env/utils.py#L89-L106)
2. [Added all encodable information of the game into classes](https://github.com/caymansimpson/poke-env/tree/master/src/poke_env/environment)
3. [Store information of all possible members of the opponents team](https://github.com/caymansimpson/poke-env/blame/master/src/poke_env/environment/double_battle.py#L418-L428)
4. [Retain memory of mons we've selected](https://github.com/caymansimpson/poke-env/blame/master/src/poke_env/environment/double_battle.py#L406-L415)
5. [Helper method to determine whether a DoubleBattleOrder is valid](https://github.com/caymansimpson/poke-env/blame/master/src/poke_env/player/battle_order.py#L154-L246)
6. [A VGC bot you can control via CLI](https://github.com/caymansimpson/reuniclusVGC/blob/main/bots/io_player.py)
7. [Max Damage Player for VGC](https://github.com/caymansimpson/reuniclusVGC/blob/main/bots/max_damage_player.py)

**Comment by hsahovic** [comment]
Hey @caymansimpson,

This is awesome. Do you mind if I integrate some of these to the core repo?

**Reply by caymansimpson** [reply]
Please go for it! LMK if I can help - I never learned how to code properly so it might be a little messy
